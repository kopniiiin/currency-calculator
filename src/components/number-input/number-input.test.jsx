import React from "react";
import {shallow} from "enzyme";

import NumberInput from "./number-input.jsx";

describe(`snapshot test: NumberInput`, () => {
  it(`should render correctly`, () => expect(
    shallow(
      <NumberInput
        name={`amount`}
        value={`4`}
        label={`amount`}
        required={false}
        onChange={() => {}}/>
    )
  ).toMatchSnapshot());
});

describe(`e2e test: NumberInput`, () => {
  it(`should call onChange with value`, () => {
    const value = `6`;
    const onChange = jest.fn();

    shallow(
      <NumberInput
        name={`amount`}
        value={`4`}
        label={`amount`}
        required={false}
        onChange={onChange}/>
    ).find(`.number-input`).simulate(`change`, {target: {value}});

    expect(onChange).toHaveBeenCalledTimes(1);
    expect(onChange.mock.calls[0][0]).toBe(value);
  });
});
