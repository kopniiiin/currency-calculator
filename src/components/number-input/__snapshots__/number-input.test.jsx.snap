// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test: NumberInput should render correctly 1`] = `
<input
  aria-label="amount"
  autoComplete="off"
  className=" number-input"
  name="amount"
  onChange={[Function]}
  pattern="(0(?=\\\\.)|[1-9])\\\\d*(\\\\.\\\\d{1,2})?"
  required={false}
  type="text"
  value="4"
/>
`;
